org-roam-extra.el

```diff
@@ -0,0 +1,268 @@
+;;; org-roam-extra.el --- Roam Research replica with Org-mode -*- coding: utf-8; lexical-binding: t -*-
+
+;; Copyright © 2020 Jethro Kuan <jethrokuan95@gmail.com>
+;; Copyright © 2020 Mykhailo Shevchuk <mail@mshevchuk.com>
+
+;; Author: Jethro Kuan <jethrokuan95@gmail.com>
+;; Contributor: Mykhailo Shevchuk <mail@mshevchuk.com>
+;; URL: https://github.com/jethrokuan/org-roam
+;; Keywords: org-mode, roam, convenience
+;; Version: 1.0.0-rc1
+;; Package-Requires: ((emacs "26.1") (dash "2.13") (f "0.17.2") (s "1.12.0") (org "9.3") (emacsql "3.0.0") (emacsql-sqlite "1.0.0"))
+
+;; This file is NOT part of GNU Emacs.
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation; either version 3, or (at your option)
+;; any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs; see the file COPYING.  If not, write to the
+;; Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+;; Boston, MA 02110-1301, USA.
+
+;;; Commentary:
+;;
+;; This library offers a tighter integration between org-roam and
+;; org-ref by providing `org-roam-extra-org-ref-notes-fn' function,
+;; which upon installing into `org-ref-notes-function' will enable
+;; org-ref to use org-roam as a backend for managing bibliography
+;; notes.
+;;
+;; Optionally, `org-roam-capture-templates' can be dynamically
+;; preformatted with bibtex field values.
+;;
+;; Optionally, automatic switching to the perspective (persp-mode)
+;; with the notes project (projectile) is possible.
+
+;;; Code:
+;;;; Library Requires
+(require 'org-roam)
+
+(autoload 'org-ref-find-bibliography "Find the bibliography in the buffer.")
+(autoload 'projectile-relevant-open-projects "projectile" "Return a list of open projects.")
+(autoload 'persp-switch "persp-mode" "Switch to the perspective with name NAME.")
+(autoload 'persp-names "persp-mode" "")
```

Comment:
Do we need to declare autoloads for externally provided functions? seems odd to me

Reply:
This is to silence the compiler about not defined functions. I don't like it either, although the idea is not mine. I'll try to figure a better way to silence the compiler and not to require these optional features unconditionally in the top level.

Reply:
Silencing the compiler is typically done with `declare-function`, rather than `autoload`.

Reply:
Thank you, I knew "Emacs had a keybinding for that"